### PR TITLE
Resolve "Processes/Calculators-Area makes it difficult to add new taks"

### LINKED ISSUE #580: Processes/Calculators-Area makes it difficult to add new taks
In GitLab by @schn_bo on Jul 27, 2023, 12:16

## What is the purpose of the change?
- In the case that so many processes have been created in the window that scrolling in the window is possible, adding new tasks is difficult due to the missing space between the last entry and the search area. (In some cases there is a space (in my opninion it depends on the window size)).

grafik

## Which tasks have to be performed?
- [ ] add spacer at the end of the list in which new task can be created vai right clicking

### PR DESCRIPTION
**Pull request by rainman110**
In GitLab by @jensschmeink on Sep 5, 2023, 09:42

## Checklist:

- [ ] A test for the new functionality was added (if applicable).
- [ ] All tests run without failure.
- [ ] The changelog has been extended, if this MR contains important changes from the users's point of view.
- [ ] The new code complies with the GTlab's style guide.
- [ ] New interface methods / functions are exported via `EXPORT`. Non-interface functions are NOT exported.
- [ ] The number of code quality warnings is not increasing.

Closes #580

### CODE REVIEW

**Comment by rainman110** [comment]
@jensschmeink What is the status of this PR? Should it be closed or are you still working on it?

**Comment by jensschmeink** [comment]
Current solution are new context menu entries to add new root task.

File: src/app/dock_widgets/process/gt_processdock.cpp [hunk 1]
```diff
@@ -919,10 +913,9 @@ GtProcessDock::makeAddMenu(QMenu& menu)
                         gtDataModel->objectFromIndex(srcIdx) : nullptr;
 
     // add empty task action
-    auto addEmptyTask = [this](GtObject* obj){
-        obj ? this->addEmptyTask(obj) : addEmptyTaskToRoot();
-    };
-    auto addemptytask = gt::gui::makeAction(tr("Empty Task"), addEmptyTask)
+    auto addEmptyTask = std::bind(&GtProcessDock::addEmptyTaskToRoot, this);
+    auto addemptytask = gt::gui::makeAction(tr("Empty Task as Root"),
```
Comment (mariusalexander) [inline src/app/dock_widgets/process/gt_processdock.cpp:917]: I would prefer "Empty Root Task" instead. Rest looks fine to me! Good job

```suggestion
    auto addemptytask = gt::gui::makeAction(tr("Empty Root Task"),
```

**Comment by jensschmeink** [comment]
@mariusalexander I applied your suggestion. Please check for approval again

**Review by mariusalexander — APPROVED** [review]

**Pull request merged**